### PR TITLE
[#127] Allow custom player markup

### PR DESCRIPTION
**Pull request by remyabel**
* [#127] Factor out option variables from template literal

* [#127] Factor out eleHTML to move into demo file

* [#127] Move eleHTML into demo file

* [#127] Add custom markup to demo

[#127] Add custom CSS for player4

[#127] Generate default APlayer if markup is empty

* [#127] Restore commented code in demo files

[#127] Enable support for Bootstrap progress bar in demo

screenshot_2017-04-02_23-09-50

**Notes**:

This approach mirrors jPlayer's approach of having the markup be in the HTML file itself.

I got lazy and left the `<ol>` generation in. Those need to be factored out too so that custom markup using Bootstrap navs can have custom markup for the playlist.

### CODE REVIEW

**Comment by liwanglin12** [comment]
@remyabel 清明之前没空 review 了..

File: src/APlayer.js [hunk 1]
```diff
@@ -154,70 +154,100 @@ class APlayer {
...
+        }
+        eleHTML += `</ol>`
+
+        this.element.querySelector('.aplayer-list').innerHTML += eleHTML;
+
+        if (this.music.pic) {
+            this.element.querySelector('.aplayer-pic').style.backgroundImage = `url('${this.music.pic}')`;
+        }
+
+        this.element.querySelector('.aplayer-icon-play').innerHTML += this.getSVG('play');
```
Comment (DIYgod) [inline src/APlayer.js]: Why? What are the advantages?

File: src/APlayer.js [hunk 2]
```diff
@@ -154,70 +154,100 @@ class APlayer {
...
+        
+        // If not using Bootstrap style, use default background
+        // TODO: find a better way than hardcoding a very specific check
+        if (!this.element.querySelector('.progress')) {
```
Comment (DIYgod) [inline src/APlayer.js]: I think we should not do it in APlayer core.

Comment (remyabel) [inline src/APlayer.js]: Thanks for the feedback. I'll close the pull request and just put these changes in my own fork.

**Pull request closed without merging**